src/main/kotlin/hr/foi/pop/backend/services/UserService.kt

```diff
@@ -41,6 +44,26 @@ class UserService {
         return user
     }
 
+    @Throws(ActivateUserException::class)
+    fun activateOrDeactivateUser(userId: String, activateUserDefinition: ActivateUserDefinitions): User {
```

Comment:
I don't like the name of this method. Methods should have single responsibility. No "ands"/"ors" in naming.
```suggestion
    fun changeUserActivationStatus(userId: String, activateUserDefinition: ActivateUserDefinitions): User {
```

However, as I've noted in an earlier comment (the one on ActivateUserDefinitions enum), it'd be better, in my opinion, to separate this in two public methods "activateUser" and "deactivateUser" which just take in a single parameter - an object of class User. Then, you could have a private function `changeUserActivationStatus` which also takes in a boolean.

Reply:
So you want me to check for user existence inside controller? Not inside service?